app/adapters/alma_adapter.rb

```diff
@@ -46,6 +46,95 @@ def get_availability_many(ids:)
     availability
   end
 
+  def get_availability_holding(id:, holding_id:, query: nil)
+    # Fetch the bib record and get the information for the individual holding
+    bibs = Alma::Bib.find(Array.wrap(id), expand: ["p_avail", "e_avail", "d_avail", "requests"].join(",")).each
+    return nil if bibs.count == 0
+
+    bib_status = AvailabilityStatus.from_bib(bib: bibs.first)
+    marc_holding = bib_status.holding(holding_id: holding_id) || {}
+
+    # Fetch the items for the holding...
+    holding_items = bib_status.holding_item_data(holding_id: holding_id, query: query)
+
+    # This is to protect in case the bib_id and the holding_id are valid but not related
+    # i.e. the holding_id is for a different bib_id than the one we received.
+    #
+    # In this case Alma will return the correct `total_count` of items for the holding_id
+    # (since the holding_id exists) but it will not return the items for the holding
+    # because the holding_id belongs to a different bib_id.
+    return nil if holding_items[:items].count == 0 && holding_items[:total_count] > 0
+
+    # ...create the availability response for each item
+    availability = []
+    holding_items[:items].each do |item|
+      item_data = item["item_data"] || {}
+      holding_data = item["holding_data"] || {}
+
+      location = item_data["location"] || {}
+      library = item_data["library"] || {}
+      in_temp_library = false
+      temp_library = {}
+      temp_location = {}
+      if holding_data["in_temp_location"]
+        in_temp_library = true
+        temp_library = holding_data["temp_library"] || {}
+        temp_location = holding_data["temp_location"] || {}
+      end
+
+      status_label = marc_holding["availability"]
+      if status_label.nil?
+        # TODO: Figure out if this is right.
+        #     When the marc_holding is nil it seems that there is a a single holding in the bib but
+        #     that holding does NOT have an ID, yet, I think for eresources that single holding
```

Comment:
I don't think we need e resources because availability is really about physical resources. Can we just ignore them somehow?

Reply:
I left the TODO comment here because this still need to be figured out. I've updated the commend to indicate that this might never be an issue for us since the Request app might not call this endpoint for electronic holdings. https://github.com/pulibrary/bibdata/blob/av_mfhd_v2/app/adapters/alma_adapter.rb#L77-L94

@hackartisan I tried to exclude the "AVE" record by not requesting "e_avail" and that made no difference (the data comes in the MARC as an "INT" tag, not "AVE :(